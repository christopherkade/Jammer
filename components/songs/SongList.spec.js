import { shallowMount, config } from '@vue/test-utils'
import cmp from './SongList.vue'
import cmpChild1 from './Song.vue'
import cmpChild2 from '@/components/Spinner.vue'

describe('[UNIT] Testing the Song component', () => {
  let wrapper

  config.mocks.$store = {
    state: {
      songs: {
        isLoading: false,
        songs: [
          {}
        ]
      }
    }
  }

  beforeEach(() => {
    wrapper = shallowMount(cmp, {
      propsData: {
        songs: [
          {}
        ]
      }
    })
  })

  it('Snapshot testing', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('Component validation', () => {
    it('should render the main (parent) component', () => {
      expect(wrapper.is(cmp)).toBe(true)
    })

    it('should render the child song component correctly', () => {
      expect(wrapper.find(cmpChild1).exists())
    })

    it('should render the child search bar component correctly', () => {
      expect(wrapper.find(cmpChild2).exists())
    })

    it('should render the songs as a list of length 1', () => {
      expect(wrapper.findAll(cmpChild1).length).toBe(1)
    })
  })
})
