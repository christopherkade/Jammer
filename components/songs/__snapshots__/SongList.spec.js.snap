// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[UNIT] Testing the Song component Snapshot testing 1`] = `
"<div><div><h1 class=\\"songs-title\\">
      My songs
    </h1> <div><song-stub song=\\"[object Object]\\"></song-stub></div></div> <!----></div>"
`;
